labs/multiverse/src/main.rs

```diff
@@ -313,6 +342,208 @@ impl App {
         }
     }
 
+    async fn wait_for_room_sync(
+        update_sender: &Sender<(bool, IndexingMessage)>,
+        sync_service: Arc<SyncService>,
+    ) -> Result<(), ()> {
+        let mut sync_subscriber = sync_service.room_list_service().state();
+
+        // Spin until there are rooms to index
+        loop {
+            match timeout(Duration::from_secs(30), sync_subscriber.next()).await {
```

Comment:
I think you'll want to apply the timeout to the `wait_for_room_sync()` method, this way we're waiting 30 seconds for each update we receive over the subscriber.